doc/example/bayes_factors.ipynb

```diff
@@ -0,0 +1,1427 @@
+    "                                                 size=1000)\n",
+    "    log_likelihoods = np.array([-m['neg_log_likelihood'](x) for x in prior_sample])\n",
+    "    m['arithmetic_log_evidence'] = logsumexp(log_likelihoods) - np.log(log_likelihoods.size)\n",
+    "    \n",
+    "    print(m['name'], f'arithmetic mean: {m[\"arithmetic_log_evidence\"]}')"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "ExecuteTime": {
+     "end_time": "2024-08-12T13:55:36.005475Z",
+     "start_time": "2024-08-12T13:55:35.934958Z"
+    }
+   },
+   "id": "ec2f000c836abad6",
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "True-Model arithmetic mean: -19.987909938105005\n",
+      "Alternative-Model arithmetic mean: -21.931602019465995\n",
+      "CPU times: user 113 ms, sys: 480 ms, total: 593 ms\n",
+      "Wall time: 68 ms\n"
+     ]
+    }
+   ],
+   "execution_count": 9
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "#### Harmonic Mean\n",
+    "\n",
+    "The harmonic mean estimator uses posterior samples to estimate the marginal likelihood:\n",
+    "\n",
+    "$$\n",
+    "P(\\mathcal{D} \\mid \\mathcal{M}) \\approx \\left( \\frac{1}{N} \\sum_{i=1}^N \\frac{1}{P(\\mathcal{D} \\mid \\theta_i, \\mathcal{M})} \\right)^{-1}\n",
+    "$$\n",
+    "\n",
+    "where $\\theta_i$ are samples from the posterior distribution.\n",
+    "\n",
+    "The harmonic mean estimator approximates the evidence from above since it tends to ignore low likelihood regions, such as those comprising the prior, leading to overestimates of the marginal likelihoods, even when asymptotically unbiased.\n",
+    "Moreover, it can be unstable and fail catastrophically. A more stable version, the stabilized harmonic mean, also uses samples from the prior (see [Newton and Raftery (1994)](https://doi.org/10.1111/j.2517-6161.1994.tb01956.x)).\n",
+    "\n",
+    "A reliable and relatively straightforward sampling method is bridge sampling (see [\"A Tutorial on Bridge Sampling\" by Gronau et al. (2017)](https://api.semanticscholar.org/CorpusID:5447695) for a nice introduction). It uses samples from a proposal and the posterior to estimate the marginal likelihood. The proposal distribution should be chosen to have a high overlap with the posterior (we construct it from half of the posterior samples by fitting a Gaussian distribution with the same mean and covariance). This method is more stable than the harmonic mean estimator, however, it accuracy may depend on the choice of the proposal distribution.\n",
+    "\n",
+    "A different approach, the learnt harmonic mean estimator, was proposed by [McEwen et al. (2021)](https://api.semanticscholar.org/CorpusID:244709474). The estimator solves the large variance problem by interpreting the harmonic mean estimator as importance sampling and introducing a new target distribution, which is learned from the posterior samples.\n"
```

Comment:
For me it is not clear what the "large variance problem" is that is solved by this approach. Maybe I overread it or it is well-known, but I am not sure what exactly this means and why this method solves it and the others don't.

Reply:
I added some more lines of explanation: the harmonic mean can have a high variance due to evaluating the likelihood at low probability regions and inverting it.